### PR TITLE
conda-lock 1.4.0, remove gitpython, update docs, add maintainer

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Notes:
- removes `gitpython` (has CVEs)
- update docs url
- clean up jinja vars
- add maintainer :wave:

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge/conda-lock ready for review :lock: :snake:

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,7 +35,6 @@ requirements:
     - clikit >=0.6.2
     - crashtest >=0.3.0
     - ensureconda >=1.3
-    - gitpython
```
[the remark this answers is not in the document]
Comment (bollwyvl) [reply recipe/meta.yaml:39]: https://nvd.nist.gov/vuln/detail/CVE-2022-24439

[the remark this answers is not in the document]
Comment (bollwyvl) [reply recipe/meta.yaml:39]: it does look like it was fixed in https://github.com/gitpython-developers/GitPython/releases/tag/3.1.30 so we could put it back with the bottom pin... but would rather see that happen upstream first.

[the remark this answers is not in the document]
Comment (bollwyvl) [reply recipe/meta.yaml:39]: (and maybe wait until the CVE has been cleared with a "fixed by" bottom pin)

[the remark this answers is not in the document]
Comment (bollwyvl) [reply recipe/meta.yaml:39]: https://github.com/conda-incubator/conda-lock/pull/318

[the remark this answers is not in the document]
Comment (mariusvniekerk) [reply recipe/meta.yaml:39]: makes sense to me.

**Review by mariusvniekerk — APPROVED** [review]

**Pull request merged**